### PR TITLE
Add dusk_ci config for nstack repo

### LINKED ISSUE #4: Add CI to the project.
We are lacking a CI for this repo. So would be nice to add github actions to check the most basic things.

### PR DESCRIPTION
**Pull request by CPerezz**
We were missing a CI for this repo. Therefore, we add it
following the reference impl that we have at `dusk-pki` repo.

Closes #4

### CODE REVIEW

File: .github/workflows/dusk_ci.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  test_stable:
+    name: Stable tests
+    runs-on: [ubuntu-latest, macos-latest]
```
Comment (ZER0) [inline .github/workflows/dusk_ci.yml:40]: Limit to `ubuntu-latest`, like we have for the rest of the jobs and repos' actions. Not quite sure why you added macos-latest, here.

Also, it seems to cause issue and the jobs is not running: 
```
No runner matching the specified labels was found: ubuntu-latest, macos-latest
```

File: .github/workflows/dusk_ci.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --release
+
+  test_nightly:
+    name: Nightly tests
+    runs-on: [ubuntu-latest, macos-latest]
```
Comment (ZER0) [inline .github/workflows/dusk_ci.yml:55]: ditto.

**Review by ZER0 — CHANGES_REQUESTED** [review]
The comments below needs to be addressed (or an explanation why is needed added, and then a fix).
Plus, as mentioned in #6 we should integrate the license header to the single file here instead having a separate PR.

File: .github/workflows/dusk_ci.yml (hunk 1 above)
Comment (CPerezz) [reply .github/workflows/dusk_ci.yml]: Yep, I wanted to try it but I'd need to look at it better.

**Review by ZER0 — APPROVED** [review]
LGTM

**Pull request merged**